### PR TITLE
Leverages testing-library as part of #1228

### LINKED ISSUE #1228: Leverage cypress-testing-library in all Cypress tests
# Feature request
We use a mixture of Cypress's 1st-class methods and [`cypress-testing-library`](https://github.com/testing-library/cypress-testing-library) methods to write our integration tests (located inside `./cypress/integration/`).

I'd like it if we ONLY used `cypress-testing-library` methods and if we never wait for requests to resolve (like [here](https://github.com/OperationCode/front-end/blob/main/cypress/integration/join.spec.js#L33)). Instead, trust the asynchronous nature of `cypress-testing-library` by using `findBy*` queries to assert for something to _eventually_ be true.

## Proposed solution
Use `cypress-testing-library` methods where able (see `resources.spec.js` and `code_schools.spec.js` for good examples).

### PR DESCRIPTION
**Pull request by logesr**
# Description of changes
Leverages cypress-testing-library in all Cypress tests. This reduces the waits used directly.

# Issue Resolved
Fixes #1228

### CODE REVIEW

File: cypress/integration/profile/update.spec.js [hunk 1]
```diff
@@ -65,8 +65,8 @@ describe(`profile/update (from login)`, () => {
     goToNextStep('Military Details');
 
     cy.get('input[name="branchOfService"]').should('have.value', 'army');
-    cy.get('input[name="yearsOfService"]').should('have.value', '3');
-    cy.get('input[name="payGrade"]').should('have.value', 'E20');
+    cy.findByLabelText('Years Of Service*').should('have.value', '3');
```
Comment (kylemh) [inline cypress/integration/profile/update.spec.js:68]: looks like you missed a few `cy.get()` in here.

File: cypress/integration/profile/update.spec.js [hunk 2]
```diff
@@ -48,15 +48,15 @@ describe(`profile/update (from login)`, () => {
     goToNextStep('Military Details');
     goToNextStep('Technology');
     cy.findByTestId(MULTI_STEP_SUBMIT_BUTTON).click();
-    cy.wait('@patchUser');
+    // cy.wait('@patchUser');
```
Comment (kylemh) [inline cypress/integration/profile/update.spec.js:51]: comment

**Review by kylemh — COMMENTED** [review]
Nearly perfect!

File: cypress/integration/profile/update.spec.js (hunk 1 above)
Comment (logesr) [reply cypress/integration/profile/update.spec.js:68]: Hi @kylemh, id is missing for several input fields. That is why I have retained few of those get() as testing-library findBy* is not working. Should we fix this in source code?

File: cypress/integration/profile/update.spec.js (hunk 2 above)
Comment (logesr) [reply cypress/integration/profile/update.spec.js:51]: I will remove this line of code in my next commit as the wait is not needed.

File: cypress/integration/profile/update.spec.js (hunk 1 above)
Comment (kylemh) [reply cypress/integration/profile/update.spec.js:68]: Feel free to make any changes to data- attributes in the source code; however, I'm curious why `findByLabelText` wouldn't work?

Comment (logesr) [reply cypress/integration/profile/update.spec.js:68]: label for attribute will refer id attribute in input field. The id attribute is missing in several fields especially select type fields. That is why it is not working here.

Comment (kylemh) [reply cypress/integration/profile/update.spec.js:68]: Ah, yes the `react-select` inputs have wild HTML output. It'd be very cool if we had a custom `cy` command for typing into a select/multiselect based on the label.

I can open a new issue for that, or you can resolve it in a separate PR if you're interested?!

**Comment by logesr** [comment]
Sure Kyle. I can give a new PR once I complete it.

File: cypress/integration/profile/update.spec.js (hunk 2 above)
Comment (logesr) [reply cypress/integration/profile/update.spec.js]: I have removed this line of code in my latest commit

File: cypress/integration/profile/update.spec.js (hunk 1 above)
Comment (logesr) [reply cypress/integration/profile/update.spec.js:67]: @kylemh I have made these changes. Will you be opening a new issue for this or shall I trigger the PR?

Comment (kylemh) [reply cypress/integration/profile/update.spec.js:67]: Feel free to resolve the problem in this PR @logesr - I can count this is 2 separate issues resolved 🙏

**Review by kylemh — APPROVED** [review]
Thanks so much for your 🎃 Hacktoberfest🎃&nbsp; contribution!

If you weren't aware, we're offering prizes too! You can read about the details [here](https://github.com/OperationCode/START_HERE#-hacktoberfest-).

Regardless, you've contributed, and we recommend you set yourself a reminder to submit the aforementioned form after November 1st (or after you meet all the criteria).

File: cypress/support/commands.js [hunk 3]
```diff
@@ -57,6 +57,18 @@ Cypress.Commands.add('checkCustomDataAttribute', (attribute, value) => {
...
+  cy.findByText(label)
+    .invoke('attr', 'for')
+    .then(name => {
+      if (options.edit) {
+        cy.get(`input#react-select-${name}-input`);
+      } else {
+        cy.get(`input[name="${name}"]`);
+      }
+    });
+});
```
Comment (kylemh) [inline cypress/support/commands.js:70]: This is awesome. I think you nailed the abstraction.

Comment (logesr) [reply cypress/support/commands.js:70]: Thanks Kyle.. :)

**Pull request merged**